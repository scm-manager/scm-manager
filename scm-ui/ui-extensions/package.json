{
  "name": "@scm-manager/ui-extensions",
  "version": "2.12.0-SNAPSHOT",
  "main": "src/index.ts",
  "license": "MIT",
  "private": false,
  "author": "Sebastian Sdorra <sebastian.sdorra@cloudogu.com>",
  "scripts": {
    "typecheck": "tsc",
    "test": "jest"
  },
  "dependencies": {
    "react": "^16.10.2"
  },
  "devDependencies": {
    "@scm-manager/babel-preset": "^2.10.1",
    "@scm-manager/eslint-config": "^2.10.1",
    "@scm-manager/jest-preset": "^2.12.1",
    "@scm-manager/prettier-config": "^2.10.1",
    "@scm-manager/tsconfig": "^2.10.1",
    "@types/enzyme": "^3.10.3",
    "@types/jest": "^24.0.19",
    "@types/react": "^16.9.9",
    "typescript": "^3.7.2"
  },
  "babel": {
    "presets": [
      "@scm-manager/babel-preset"
    ]
  },
  "jest": {
    "preset": "@scm-manager/jest-preset"
  },
  "prettier": "@scm-manager/prettier-config",
  "publishConfig": {
    "access": "public"
  }
}
