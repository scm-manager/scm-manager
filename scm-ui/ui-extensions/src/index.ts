/*
 * Copyright (c) 2020 - present Cloudogu GmbH
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

export { default as binder, Binder, ExtensionPointDefinition } from "./binder";
export * from "./useBinder";
export {
  default as ExtensionPoint,
  RenderableExtensionPointDefinition,
  SimpleRenderableDynamicExtensionPointDefinition,
} from "./ExtensionPoint";
export { default as ExtractProps } from "./extractProps";

// suppress eslint prettier warning,
// because prettier does not understand "* as"
// eslint-disable-next-line prettier/prettier
export * as extensionPoints from "./extensionPoints";
