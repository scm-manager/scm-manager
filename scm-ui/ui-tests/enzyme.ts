/*
 * Copyright (c) 2020 - present Cloudogu GmbH
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

import "raf/polyfill";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

// Temporary hack to suppress error
// https://github.com/facebook/create-react-app/issues/3199#issuecomment-345024029
window.requestAnimationFrame = function(callback) {
  setTimeout(callback, 0);
  return 0;
};

Enzyme.configure({
  adapter: new Adapter()
});
