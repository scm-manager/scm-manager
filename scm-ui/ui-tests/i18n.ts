/*
 * MIT License
 *
 * Copyright (c) 2020-present Cloudogu GmbH and Contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

export const jestMock = jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => (Component: any) => {
    Component.defaultProps = {
      ...Component.defaultProps,
      t: (key: string) => key
    };
    return Component;
  },
  useTranslation: (ns: string) => {
    return [(key: string) => key];
  }
}));
