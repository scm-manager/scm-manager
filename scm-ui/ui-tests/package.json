{
  "name": "@scm-manager/ui-tests",
  "version": "2.12.0-SNAPSHOT",
  "description": "UI-Tests helpers",
  "author": "Sebastian Sdorra <sebastian.sdorra@cloudogu.com>",
  "license": "MIT",
  "private": false,
  "scripts": {
    "typecheck": "tsc"
  },
  "dependencies": {
    "babel-plugin-istanbul": "^5.2.0",
    "enzyme": "^3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "enzyme-context": "^1.1.2",
    "enzyme-context-react-router-4": "^2.0.0",
    "jest": "^26.0.0",
    "raf": "^3.4.1",
    "react-test-renderer": "^16.10.2"
  },
  "prettier": "@scm-manager/prettier-config",
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@types/enzyme": "^3.10.3",
    "@types/enzyme-adapter-react-16": "^1.0.5",
    "@types/jest": "^24.0.19",
    "typescript": "^3.7.2"
  }
}
