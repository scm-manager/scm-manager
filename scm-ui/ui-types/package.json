{
  "name": "@scm-manager/ui-types",
  "version": "2.12.0-SNAPSHOT",
  "description": "Flow types for SCM-Manager related Objects",
  "main": "src/index.ts",
  "files": [
    "dist",
    "src"
  ],
  "repository": "https://bitbucket.org/sdorra/scm-manager",
  "author": "Sebastian Sdorra <sebastian.sdorra@cloudogu.com>",
  "license": "MIT",
  "scripts": {
    "typecheck": "tsc"
  },
  "devDependencies": {
    "typescript": "^3.7.2"
  },
  "babel": {
    "presets": [
      "@scm-manager/babel-preset"
    ]
  },
  "prettier": "@scm-manager/prettier-config",
  "publishConfig": {
    "access": "public"
  }
}
