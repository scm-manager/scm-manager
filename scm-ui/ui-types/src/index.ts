/*
 * MIT License
 *
 * Copyright (c) 2020-present Cloudogu GmbH and Contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

export { Action } from "./Action";
export { Link, Links, Collection, PagedCollection } from "./hal";

export { Me } from "./Me";
export { DisplayedUser, User } from "./User";
export { Group, Member } from "./Group";

export {
  Repository,
  RepositoryCollection,
  RepositoryGroup,
  RepositoryCreation,
  Namespace,
  NamespaceCollection,
  RepositoryUrlImport
} from "./Repositories";
export { RepositoryType, RepositoryTypeCollection } from "./RepositoryTypes";

export { Branch, BranchRequest } from "./Branches";

export { Person } from "./Person";

export { Changeset, Contributor, ParentChangeset } from "./Changesets";

export { Signature } from "./Signature";

export { AnnotatedSource, AnnotatedLine } from "./Annotate";

export { Tag } from "./Tags";

export { Config, AnonymousMode } from "./Config";

export { IndexResources } from "./IndexResources";

export { Permission, PermissionCreateEntry, PermissionCollection } from "./RepositoryPermissions";

export { SubRepository, File } from "./Sources";

export { SelectValue, AutocompleteObject } from "./Autocomplete";

export { Plugin, PluginCollection, PluginGroup, PendingPlugins } from "./Plugin";

export { RepositoryRole } from "./RepositoryRole";

export { NamespaceStrategies } from "./NamespaceStrategies";
