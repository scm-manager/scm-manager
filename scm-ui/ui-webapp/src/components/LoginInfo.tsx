/*
 * MIT License
 *
 * Copyright (c) 2020-present Cloudogu GmbH and Contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
import React, { FC } from "react";
import InfoBox from "./InfoBox";
import LoginForm from "./LoginForm";
import { Image, Loading } from "@scm-manager/ui-components";
import { ExtensionPoint, extensionPoints } from "@scm-manager/ui-extensions";
import styled from "styled-components";
import { useLoginInfo } from "@scm-manager/ui-api";
import { useTranslation } from "react-i18next";

const TopMarginBox = styled.div`
  margin-top: 5rem;
`;

const AvatarWrapper = styled.figure`
  display: flex;
  justify-content: center;
  margin: -70px auto 20px;
  width: 128px;
  height: 128px;
  background: var(--scm-white-color);
  border: 1px solid lightgray;
  border-radius: 50%;
`;

const AvatarImage = styled(Image)`
  width: 75%;
  margin-left: 0.25rem;
  padding: 5px;
`;

type Props = {
  /**
   * @deprecated Unused because the component now uses {@link useLoginInfo} internally.
   */
  loginInfoLink?: string;
  loading?: boolean;
  error?: Error | null;
  loginHandler: (username: string, password: string) => void;
};

const LoginInfo: FC<Props> = (props) => {
  const { isLoading: isLoadingLoginInfo, data: info } = useLoginInfo();
  const [t] = useTranslation("commons");

  if (isLoadingLoginInfo) {
    return <Loading />;
  }

  let infoPanel;
  if (info) {
    infoPanel = (
      <div className="column is-7 is-offset-1 p-0">
        <InfoBox item={info.feature} type="feature" />
        <InfoBox item={info.plugin} type="plugin" />
      </div>
    );
  }

  return (
    <>
      <div className="column is-4 box has-text-centered has-background-secondary-less">
        <h3 className="title">{t("login.title")}</h3>
        <p className="subtitle">{t("login.subtitle")}</p>
        <TopMarginBox className="box">
          <AvatarWrapper>
            <AvatarImage src="/images/blibSmallLightBackground.svg" alt={t("login.logo-alt")} />
          </AvatarWrapper>
          <ExtensionPoint<extensionPoints.LoginForm> name="login.form" props={{}}>
            <LoginForm {...props} />
          </ExtensionPoint>
        </TopMarginBox>
      </div>
      {infoPanel}
    </>
  );
};

export default LoginInfo;
