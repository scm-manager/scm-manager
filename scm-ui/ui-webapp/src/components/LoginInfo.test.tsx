/*
 * MIT License
 *
 * Copyright (c) 2020-present Cloudogu GmbH and Contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { LoginInfo as LoginInfoType } from "@scm-manager/ui-types";

import "@scm-manager/ui-tests";
import LoginInfo from "./LoginInfo";
import { Binder, BinderContext, extensionPoints } from "@scm-manager/ui-extensions";

jest.mock("@scm-manager/ui-api", () => ({
  useLoginInfo: jest.fn(() => ({
    isLoading: false,
    data: {
      _links: {},
      feature: { title: "Test", summary: "Test" },
      plugin: { summary: "Test", title: "Test" },
    } as LoginInfoType,
  })),
  urls: {
    getPageFromMatch: jest.fn(() => 1),
    withContextPath: jest.fn((it) => it),
  },
}));

describe("LoginInfo", () => {
  const withBinder = (ui, binder) => <BinderContext.Provider value={binder}>{ui}</BinderContext.Provider>;

  it("should render login page", () => {
    const loginHandler = jest.fn();
    const binder = new Binder("test");
    const reactTestRenderer = TestRenderer.create(withBinder(<LoginInfo loginHandler={loginHandler} />, binder));
    expect(reactTestRenderer.toJSON()).toMatchSnapshot();
  });

  it("should render extension", () => {
    const loginHandler = jest.fn();
    const binder = new Binder("test");
    binder.bind<extensionPoints.LoginForm>("login.form", () => <button>Login with OAuth2</button>);
    const reactTestRenderer = TestRenderer.create(withBinder(<LoginInfo loginHandler={loginHandler} />, binder));
    expect(reactTestRenderer.toJSON()).toMatchSnapshot();
  });

  it("should render extension with login form", () => {
    const loginHandler = jest.fn();
    const binder = new Binder("test");
    binder.bind<extensionPoints.LoginForm>("login.form", ({ children }) => (
      <>
        {children}
        <button>Login with OAuth2</button>
      </>
    ));
    const reactTestRenderer = TestRenderer.create(withBinder(<LoginInfo loginHandler={loginHandler} />, binder));
    expect(reactTestRenderer.toJSON()).toMatchSnapshot();
  });
});
