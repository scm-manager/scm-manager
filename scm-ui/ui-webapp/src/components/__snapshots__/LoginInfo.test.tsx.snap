// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginInfo should render extension 1`] = `
Array [
  <div
    className="column is-4 box has-text-centered has-background-secondary-less"
  >
    <h3
      className="title"
    >
      login.title
    </h3>
    <p
      className="subtitle"
    >
      login.subtitle
    </p>
    <div
      className="LoginInfo__TopMarginBox-sc-dmmy9i-0 hXYZcL box"
    >
      <figure
        className="LoginInfo__AvatarWrapper-sc-dmmy9i-1 ikoLxp"
      >
        <img
          alt="login.logo-alt"
          className="LoginInfo__AvatarImage-sc-dmmy9i-2 gFcMrK"
          src="/images/blibSmallLightBackground.svg"
        />
      </figure>
      <button>
        Login with OAuth2
      </button>
    </div>
  </div>,
  <div
    className="column is-7 is-offset-1 p-0"
  >
    <a
      className="is-block mb-5"
    >
      <div
        className="InfoBox__InfoBoxWrapper-sc-qh4uzj-2 dgRYgc box media"
      >
        <figure
          className="media-left"
        >
          <div
            className="InfoBox__FixedSizedIconWrapper-sc-qh4uzj-0 bCVWB image box has-text-weight-bold has-text-secondary-least has-background-info is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
          >
            <i
              aria-label=""
              className="fas fa-fw fa-star has-text-white mb-2 fa-2x"
              onKeyPress={[Function]}
            />
            <div
              className="is-size-4 has-text-white"
            >
              login.feature
            </div>
            <div
              className="is-size-4 has-text-white"
            >
              login.tip
            </div>
          </div>
        </figure>
        <div
          className="InfoBox__ContentWrapper-sc-qh4uzj-1 eiUgVo media-content content ml-5"
        >
          <h4
            className="has-text-link"
          >
            Test
          </h4>
          <p>
            Test
          </p>
        </div>
      </div>
    </a>
    <a
      className="is-block mb-5"
    >
      <div
        className="InfoBox__InfoBoxWrapper-sc-qh4uzj-2 dgRYgc box media"
      >
        <figure
          className="media-left"
        >
          <div
            className="InfoBox__FixedSizedIconWrapper-sc-qh4uzj-0 bCVWB image box has-text-weight-bold has-text-secondary-least has-background-info is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
          >
            <i
              aria-label=""
              className="fas fa-fw fa-puzzle-piece has-text-white mb-2 fa-2x"
              onKeyPress={[Function]}
            />
            <div
              className="is-size-4 has-text-white"
            >
              login.plugin
            </div>
            <div
              className="is-size-4 has-text-white"
            >
              login.tip
            </div>
          </div>
        </figure>
        <div
          className="InfoBox__ContentWrapper-sc-qh4uzj-1 eiUgVo media-content content ml-5"
        >
          <h4
            className="has-text-link"
          >
            Test
          </h4>
          <p>
            Test
          </p>
        </div>
      </div>
    </a>
  </div>,
]
`;

exports[`LoginInfo should render extension with login form 1`] = `
Array [
  <div
    className="column is-4 box has-text-centered has-background-secondary-less"
  >
    <h3
      className="title"
    >
      login.title
    </h3>
    <p
      className="subtitle"
    >
      login.subtitle
    </p>
    <div
      className="LoginInfo__TopMarginBox-sc-dmmy9i-0 hXYZcL box"
    >
      <figure
        className="LoginInfo__AvatarWrapper-sc-dmmy9i-1 ikoLxp"
      >
        <img
          alt="login.logo-alt"
          className="LoginInfo__AvatarImage-sc-dmmy9i-2 gFcMrK"
          src="/images/blibSmallLightBackground.svg"
        />
      </figure>
      <form
        onSubmit={[Function]}
      >
        <fieldset
          className="field"
        >
          
          <div
            className="control"
          >
            <input
              aria-describedby="input_5"
              aria-labelledby="input_4"
              className="input"
              data-testid="username-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="login.username-placeholder"
            />
          </div>
        </fieldset>
        <fieldset
          className="field"
        >
          
          <div
            className="control"
          >
            <input
              aria-describedby="input_7"
              aria-labelledby="input_6"
              className="input"
              data-testid="password-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="login.password-placeholder"
              type="password"
            />
          </div>
        </fieldset>
        <button
          className="button is-primary is-fullwidth"
          data-testid="login-button"
          onClick={[Function]}
          type="submit"
        >
          <span>
             
            login.submit
             
          </span>
        </button>
      </form>
      <button>
        Login with OAuth2
      </button>
    </div>
  </div>,
  <div
    className="column is-7 is-offset-1 p-0"
  >
    <a
      className="is-block mb-5"
    >
      <div
        className="InfoBox__InfoBoxWrapper-sc-qh4uzj-2 dgRYgc box media"
      >
        <figure
          className="media-left"
        >
          <div
            className="InfoBox__FixedSizedIconWrapper-sc-qh4uzj-0 bCVWB image box has-text-weight-bold has-text-secondary-least has-background-info is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
          >
            <i
              aria-label=""
              className="fas fa-fw fa-star has-text-white mb-2 fa-2x"
              onKeyPress={[Function]}
            />
            <div
              className="is-size-4 has-text-white"
            >
              login.feature
            </div>
            <div
              className="is-size-4 has-text-white"
            >
              login.tip
            </div>
          </div>
        </figure>
        <div
          className="InfoBox__ContentWrapper-sc-qh4uzj-1 eiUgVo media-content content ml-5"
        >
          <h4
            className="has-text-link"
          >
            Test
          </h4>
          <p>
            Test
          </p>
        </div>
      </div>
    </a>
    <a
      className="is-block mb-5"
    >
      <div
        className="InfoBox__InfoBoxWrapper-sc-qh4uzj-2 dgRYgc box media"
      >
        <figure
          className="media-left"
        >
          <div
            className="InfoBox__FixedSizedIconWrapper-sc-qh4uzj-0 bCVWB image box has-text-weight-bold has-text-secondary-least has-background-info is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
          >
            <i
              aria-label=""
              className="fas fa-fw fa-puzzle-piece has-text-white mb-2 fa-2x"
              onKeyPress={[Function]}
            />
            <div
              className="is-size-4 has-text-white"
            >
              login.plugin
            </div>
            <div
              className="is-size-4 has-text-white"
            >
              login.tip
            </div>
          </div>
        </figure>
        <div
          className="InfoBox__ContentWrapper-sc-qh4uzj-1 eiUgVo media-content content ml-5"
        >
          <h4
            className="has-text-link"
          >
            Test
          </h4>
          <p>
            Test
          </p>
        </div>
      </div>
    </a>
  </div>,
]
`;

exports[`LoginInfo should render login page 1`] = `
Array [
  <div
    className="column is-4 box has-text-centered has-background-secondary-less"
  >
    <h3
      className="title"
    >
      login.title
    </h3>
    <p
      className="subtitle"
    >
      login.subtitle
    </p>
    <div
      className="LoginInfo__TopMarginBox-sc-dmmy9i-0 hXYZcL box"
    >
      <figure
        className="LoginInfo__AvatarWrapper-sc-dmmy9i-1 ikoLxp"
      >
        <img
          alt="login.logo-alt"
          className="LoginInfo__AvatarImage-sc-dmmy9i-2 gFcMrK"
          src="/images/blibSmallLightBackground.svg"
        />
      </figure>
      <form
        onSubmit={[Function]}
      >
        <fieldset
          className="field"
        >
          
          <div
            className="control"
          >
            <input
              aria-describedby="input_1"
              aria-labelledby="input_0"
              className="input"
              data-testid="username-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="login.username-placeholder"
            />
          </div>
        </fieldset>
        <fieldset
          className="field"
        >
          
          <div
            className="control"
          >
            <input
              aria-describedby="input_3"
              aria-labelledby="input_2"
              className="input"
              data-testid="password-input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="login.password-placeholder"
              type="password"
            />
          </div>
        </fieldset>
        <button
          className="button is-primary is-fullwidth"
          data-testid="login-button"
          onClick={[Function]}
          type="submit"
        >
          <span>
             
            login.submit
             
          </span>
        </button>
      </form>
    </div>
  </div>,
  <div
    className="column is-7 is-offset-1 p-0"
  >
    <a
      className="is-block mb-5"
    >
      <div
        className="InfoBox__InfoBoxWrapper-sc-qh4uzj-2 dgRYgc box media"
      >
        <figure
          className="media-left"
        >
          <div
            className="InfoBox__FixedSizedIconWrapper-sc-qh4uzj-0 bCVWB image box has-text-weight-bold has-text-secondary-least has-background-info is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
          >
            <i
              aria-label=""
              className="fas fa-fw fa-star has-text-white mb-2 fa-2x"
              onKeyPress={[Function]}
            />
            <div
              className="is-size-4 has-text-white"
            >
              login.feature
            </div>
            <div
              className="is-size-4 has-text-white"
            >
              login.tip
            </div>
          </div>
        </figure>
        <div
          className="InfoBox__ContentWrapper-sc-qh4uzj-1 eiUgVo media-content content ml-5"
        >
          <h4
            className="has-text-link"
          >
            Test
          </h4>
          <p>
            Test
          </p>
        </div>
      </div>
    </a>
    <a
      className="is-block mb-5"
    >
      <div
        className="InfoBox__InfoBoxWrapper-sc-qh4uzj-2 dgRYgc box media"
      >
        <figure
          className="media-left"
        >
          <div
            className="InfoBox__FixedSizedIconWrapper-sc-qh4uzj-0 bCVWB image box has-text-weight-bold has-text-secondary-least has-background-info is-flex is-flex-direction-column is-justify-content-center is-align-items-center"
          >
            <i
              aria-label=""
              className="fas fa-fw fa-puzzle-piece has-text-white mb-2 fa-2x"
              onKeyPress={[Function]}
            />
            <div
              className="is-size-4 has-text-white"
            >
              login.plugin
            </div>
            <div
              className="is-size-4 has-text-white"
            >
              login.tip
            </div>
          </div>
        </figure>
        <div
          className="InfoBox__ContentWrapper-sc-qh4uzj-1 eiUgVo media-content content ml-5"
        >
          <h4
            className="has-text-link"
          >
            Test
          </h4>
          <p>
            Test
          </p>
        </div>
      </div>
    </a>
  </div>,
]
`;
